core/src/main/java/org/mule/processor/IdempotentRedeliveryPolicy.java

```diff
@@ -285,10 +285,16 @@ private String getIdForEvent(MuleEvent event) throws Exception
     {
         if (useSecureHash)
         {
+            byte[] bytes;
             MuleMessage message = event.getMessage();
             Object payload = message.getPayload();
-            byte[] payloadBytes = (byte[]) objectToByteArray.transform(payload);
-            byte[] bytes = concat(payloadBytes, message.getUniqueId().getBytes());
+            if(Boolean.parseBoolean(System.getProperty("RollBackExceptionStrategy","false")))
+            {
+                byte[] payloadBytes = (byte[]) objectToByteArray.transform(payload);
+                bytes = concat(payloadBytes, message.getUniqueId().getBytes());
+            } else {
```

Comment:
```suggestion
            } 
            else 
            {
```
Yeah, super bothersome but we should be consistent with the ugly use of braces in mule 3